pkg/storage/cedar_system_bookmark.go

```diff
@@ -0,0 +1,83 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// CreateCedarSystemBookmark creates a new cedar system bookmark object in the database
+func (s *Store) CreateCedarSystemBookmark(ctx context.Context, cedarSystemBookmark *models.CedarSystemBookmark) (*models.CedarSystemBookmark, error) {
+	euaUserID := appcontext.Principal(ctx).ID()
+	createAt := s.clock.Now().UTC()
+
+	cedarSystemBookmark.CreatedAt = &createAt
+	cedarSystemBookmark.EUAUserID = euaUserID
+	const createCedarSystemBookmarkSQL = `
+		INSERT INTO cedar_system_bookmarks (
+			eua_user_id,
+			cedar_system_id,
+			created_at
+		)
+		VALUES (
+			:eua_user_id,
+			:cedar_system_id,
+			:created_at
+		)`
+	_, err := s.db.NamedExecContext(
+		ctx,
+		createCedarSystemBookmarkSQL,
+		cedarSystemBookmark,
+	)
+	if err != nil {
+		appcontext.ZLogger(ctx).Error("Failed to create cedar system bookmark with error %s", zap.Error(err))
```

Comment:
Do we potentially want some error handling specific to the case where the bookmark already exists? Trying to (re)create an existing bookmark doesn't really seem like an error to me. I'm thinking about a situation where a user accidentally double-sends their request to create a bookmark from the frontend, then receives an error from the second request, when really we should either ignore the second request or just updated the `created_at` column.

Reply:
Yeah - adding a bookmark could be viewed as an idempotent operation. In playing around locally though, I'm having a hard time actually getting the error code 😓

Reply:
Got it - if you import `"github.com/lib/pq"` and log `fmt.Println("ERROR WAS", err.(*pq.Error).Code)` in this block, you'll see that it resolves to `23505` - which is `unique_violation` [in this table](https://www.postgresql.org/docs/9.2/errcodes-appendix.html)

Reply:
> Yeah - adding a bookmark could be viewed as an idempotent operation. In playing around locally though, I'm having a hard time actually getting the error code 😓

I removed it in a previous commit

Reply:
I think if that code is found there's a bit more logic, since returning `cedarSystemBookmark` here would be misleading, as the `CreatedAt` would be `Now()`, and not whatever it was in the DB 🤔

Reply:
Actually - looks like modifying the SQL to this does the trick

```
	const createCedarSystemBookmarkSQL = `
		INSERT INTO cedar_system_bookmarks (
			eua_user_id,
			cedar_system_id,
			created_at
		)
		VALUES (
			:eua_user_id,
			:cedar_system_id,
			:created_at
		) ON CONFLICT ON CONSTRAINT cedar_system_bookmarks_pkey DO UPDATE SET created_at = :created_at`
		```

Reply:
@ClayBenson94 What if the query instead did an upsert, where if the bookmark already exists it updates `created_at` to the current time?

Reply:
Yes, what Clay's edit proposes.

Reply:
Looks good as of https://github.com/CMSgov/easi-app/pull/1419/commits/d4f1c3110a12d0f18388d54c470850d74ba7abc3.